iceoryx_hoofs/include/iceoryx_hoofs/cxx/variant.hpp

```diff
@@ -54,6 +55,26 @@ struct in_place_type
     using type = T;
 };
 
+template <typename N>
```

Comment:
This is a construct which should be never used directly by the user. Could you therefore please move `is_in_place_index` and `is_in_place_type` to `variant_internal.hpp` into the `internal` namespace.